pkg/devicetwin/dtcontext/dtcontext_test.go

```diff
@@ -0,0 +1,490 @@
+				t.Errorf("NodeID = %v, Want = %v", got.NodeID, test.want.NodeID)
+				return
+			}
+			if !reflect.DeepEqual(got.CommChan, test.want.CommChan) {
+				t.Errorf("CommunicationChannel = %v, Want =%v", got.CommChan, test.want.CommChan)
+				return
+			}
+			if cap(got.ConfirmChan) != cap(test.want.ConfirmChan) {
+				t.Errorf("ConfirmChan size = %v, Want = %v", cap(got.ConfirmChan), cap(test.want.ConfirmChan))
+				return
+			}
+			if !reflect.DeepEqual(got.ModulesContext, test.want.ModulesContext) {
+				t.Errorf("ModulesContext = %v, Want = %v", got.ModulesContext, test.want.ModulesContext)
+				return
+			}
+			if !reflect.DeepEqual(got.State, test.want.State) {
+				t.Errorf("State = %v, Want = %v", got.State, test.want.State)
+				return
+			}
+		})
+	}
+}
+
+//TestCommTo is function to test CommTo().
+func TestCommTo(t *testing.T) {
+	commChan := make(map[string]chan interface{})
+	testInterface := make(chan interface{}, 1)
+	commChan["test"] = testInterface
+	dtContext := &DTContext{
+		CommChan: commChan,
+	}
+	var returnValue interface{}
+	tests := []struct {
+		name       string
+		modulename string
+		content    interface{}
+		wantErr    error
+	}{
+		{
+			//Failure Case
+			name:       "ModuleNotPresent",
+			modulename: "ModuleA",
+			content:    nil,
+			wantErr:    errors.New("Not found chan to communicate"),
+		},
+		{
+			//Success Case
+			name:       "ModulePresent",
+			modulename: "test",
+			content:    dttype.MembershipUpdate{AddDevices: []dttype.Device{{ID: "DeviceA", Name: "Router", State: "unknown"}}},
+			wantErr:    nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			err := dtContext.CommTo(test.modulename, test.content)
+			if err == nil {
+				returnValue = <-testInterface
+			} else {
+				returnValue = nil
+			}
+			if !reflect.DeepEqual(err, test.wantErr) {
+				t.Errorf("Error in CommTo %v, Want %v", err, test.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(returnValue, test.content) {
+				t.Errorf("Got %v on channel, Want % v on Channel", returnValue, test.content)
+				return
+			}
+		})
+	}
+}
+
+//TestHeartBeat is function to test HeartBeat().
+func TestHeartBeat(t *testing.T) {
+	tests := []struct {
+		name       string
+		dtc        *DTContext
+		moduleName string
+		content    interface{}
+		wantBool   bool
+	}{
+		{
+			//Success Case
+			name: "PingTest",
+			dtc: &DTContext{
+				ModulesHealth: &sync.Map{},
+			},
+			moduleName: "ModuleA",
+			content:    "ping",
+			wantBool:   false,
+		},
+		{
+			//Failure Case
+			name: "StopTest",
+			dtc: &DTContext{
+				ModulesHealth: &sync.Map{},
+			},
+			moduleName: "ModuleB",
+			content:    "stop",
+			wantBool:   true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			dtc := &DTContext{
+				ModulesHealth: test.dtc.ModulesHealth,
+			}
+			if err := dtc.HeartBeat(test.moduleName, test.content); (err != nil) != test.wantBool {
```

Comment:
Can we assert the error here. This will help in checking whether the function call failed with correct error.
Applicable to all similar cases below